contracts/modules/RelayerManager.sol

```diff
@@ -387,25 +371,26 @@ abstract contract RelayerManager is BaseModule {
         assembly { chainId := chainid() }
     }
 
-    function getRequiredSignaturesToStartSession(address _wallet) internal view returns (uint256, OwnerSignature) {
-        // owner  + [n/2] guardians
-        uint numberOfGuardians = Utils.ceil(guardianStorage.guardianCount(_wallet), 2);
-        require(numberOfGuardians > 0, "RM: no guardians set on wallet");
-        uint numberOfSignatures = 1 + numberOfGuardians;
-        return (numberOfSignatures, OwnerSignature.Required);
+    function startSession(address _wallet, address _sessionUser, uint64 _duration) external 
+    onlyWallet(_wallet) returns (bool)
+    {
+        require(_sessionUser != address(0), "RM: Invalid session user");
+        require(_duration > 0, "RM: Invalid session duration");
+
+        uint64 expiry = Utils.safe64(block.timestamp + _duration);
```

Comment:
I'm not sure if basing the expiration of a session on the current timestamp is a good idea in terms of security. What happens if I'm starting an 1 hour session but the transaction takes 30 minutes to be mined... I may end up having an open session when I didn't want to. Maybe more secure to have the expiry absolute?

Reply:
I changed this logic for a similar reason that if a block takes 30 min to mine and I want my session to be 30 min long then it would have expired by the time it's committed. I think it's going to be more difficult and error prone if we make the client adjust session lengths based on current block times rather than the opposite.